### ISSUE TITLE
♻️ Refactor data structure to smaller chunks

### ISSUE THREAD

**Issue by sthiepaan** [bhatvikrant/world-countries-capitals#63]
After working on few features/properties for this project, I noticed that it is pretty hard to work with especially when it comes to write tests for API. When we want to compare output of some _API methods_, we need to duplicate content of `data.json` file. Another thing is that currently we have _JSON_ file (with almost `5000` lines 😱) and whenever we add/update/remove some property there, it is a bit annoying and hard to check those changes on _code review_.

What we could do is refactor _JSON_ data file by splitting it to multiple _JavaScript_ files and then using imports/exports to structure it. This way, we can achieve the same thing (having whole data in one) but also we could "build" smaller chunks for testing purposes (instead of duplicating data).

I have some concept how might it look like, but first I would like to see what you think.
Feel free to share your ideas and have an input how we could achieve this!

**Comment by bhatvikrant** [comment]
Yes, @sthiepaan I agree with you completely. We need to modularize the codebase. 
Testing and PR reviewing processes are kinda hard right now, no doubt about that.  

What do you have in mind? Let's list out what all we can do to modularize the codebase.

**Comment by sthiepaan** [comment]
What we could do at first is basically:
1. Split `data.json` to separate _JavaScript_ files for each country individually
In this case what we could do is for example something like:
```
...
├  data
├ eu
│ ├ pl.js
│ └ de.js
├ af
│ └ zw.js
...
```
This allow us to focus only on a specific country or countries for specific continent. We still need to provide properties for all countries to have consistent _data model_, but I think this could be solved in another way (e.g. _Node.js_ task to add new property to all countries).
The downside of it is that whenever we populate the data, instead of working on a single file, we have to modify multiple files (which should not be a really big deal tho). On the other hand, as we mentioned earlier, the _code review_ is pretty annoying at the moment.

2. Split `index.js` to separate _JS_ files for each _API_ Method
Splitted files, less confusion and whenever we want to add/change/remove a new method, we will not struggle with that many conflicts that we had so far.

3. Split `getCountriesSpec.js` to separate _JS_ files for each _API_ Test
Same as point 2. but for tests. Also, together with 1., this will allow us to reuse data model to compare output of the methods together with different combinations of the _data models_ that we have (no more data repetitions across the project).

---

In general, we can split the problem into multiple smaller parts. For instance, points 2. and 3. can be done separately.
This way we do not make too big _Pull Request_, so on the _code review_ we can focus just on one aspect of refactoring.
